app/addons/config/resources.js

```diff
@@ -30,20 +31,20 @@ Config.OptionModel = Backbone.Model.extend({
   isNew () { return false; },
 
   sync (method, model) {
-
-    var params = {
-      url: model.url(),
-      contentType: 'application/json',
-      dataType: 'json',
-      data: JSON.stringify(model.get('value'))
-    };
-
+    let operation = put;
     if (method === 'delete') {
-      params.type = 'DELETE';
-    } else {
-      params.type = 'PUT';
+      operation = deleteRequest;
     }
-    return $.ajax(params);
+
+    return operation(
+      model.url(),
+      model.get('value')
```

Comment:
Passing the value for the deleteRequest might have side effects. deleteRequest take the second parameter as options (not body)

Reply:
Good catch. I changed the code since the body is not needed for the DELETE request.